examples/documentation_examples/examples.go

```diff
@@ -2956,3 +2957,188 @@ func StableAPIExamples() {
 	StableAPINonStrictExample()
 	StableAPIDeprecationErrorsExample()
 }
+
+func insertSnapshotQueryTestData(ctx context.Context, t *testing.T, client *mongo.Client) {
+	_, err := client.Database("pets").Collection("cats").InsertMany(ctx, []interface{}{
+		bson.D{
+			{"adoptable", false},
+			{"name", "Miyagi"},
+			{"color", "grey-white"},
+			{"age", 14},
+		},
+		bson.D{
+			{"adoptable", true},
+			{"name", "Joyce"},
+			{"color", "black"},
+			{"age", 10},
+		},
+	})
+	require.NoError(t, err)
+
+	_, err = client.Database("pets").Collection("dogs").InsertMany(ctx, []interface{}{
+		bson.D{
+			{"adoptable", true},
+			{"name", "Cormac"},
+			{"color", "rust"},
+			{"age", 7},
+		},
+		bson.D{
+			{"adoptable", true},
+			{"name", "Frank"},
+			{"color", "yellow"},
+			{"age", 2},
+		},
+	})
+	require.NoError(t, err)
+
+	_, err = client.Database("retail").Collection("sales").InsertMany(ctx, []interface{}{
+		bson.D{
+			{"shoeType", "hiking boot"},
+			{"price", 30.0},
+			{"saleDate", time.Now()},
+		},
+	})
+	require.NoError(t, err)
+}
+
+func rollbackSnapshotQueryTestData(ctx context.Context, client *mongo.Client) {
+	client.Database("pets").Drop(ctx)
+	client.Database("retail").Drop(ctx)
+}
+
+func snapshotQueryPetExample(ctx context.Context, t *testing.T, client *mongo.Client) error {
+	// Start Snapshot Query Example 1
+	// Connect to the pets database
+	db := client.Database("pets")
+
+	// Start a session with snapshotting
+	sess, err := client.StartSession(new(options.SessionOptions).SetSnapshot(true))
+	if err != nil {
+		return err
+	}
+	defer sess.EndSession(ctx)
+
+	var adoptablePetsCount int32
+	err = mongo.WithSession(ctx, sess, func(sc mongo.SessionContext) error {
+		// count the adoptable cats
+		adoptableCatsOuput := "adoptableCatsCount"
+		cursor, err := db.Collection("cats").Aggregate(sc, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableCatsOuput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(sc)
+		resp, err := cursor.Current.LookupErr(adoptableCatsOuput)
+		if err != nil {
+			return err
+		}
+		adoptableCatsCount, _ := resp.Int32OK()
+		adoptablePetsCount += adoptableCatsCount
+
+		// count the adoptable dogs
+		adoptableDogsOuput := "adoptableDogsCount"
+		cursor, err = db.Collection("dogs").Aggregate(sc, mongo.Pipeline{
+			bson.D{{"$match", bson.D{{"adoptable", true}}}},
+			bson.D{{"$count", adoptableDogsOuput}},
+		})
+		if err != nil {
+			return err
+		}
+
+		cursor.Next(sc)
+		resp, err = cursor.Current.LookupErr(adoptableDogsOuput)
+		if err != nil {
+			return err
+		}
+		adoptableDogsCount, _ := resp.Int32OK()
```

Comment:
Same as the comment above: This can be simplified some and we should handle the `ok` value.

E.g.
```go
adoptableDogsCount, ok := cursor.Current.Lookup(adoptableDogsOuput).Int32OK()
if !ok {
	return fmt.Errorf("failed to find int32 field %q in document %v", adoptableDogsOuput, cursor.Current)
}
```